lib/src/main/java/com/auth0/jwt/JWTVerifier.java

```diff
@@ -51,43 +46,54 @@ static Verification init(Algorithm algorithm) throws IllegalArgumentException {
 
     public static class BaseVerification implements Verification {
         private final Algorithm algorithm;
-        private final Map<String, Object> expectedClaims;
+        private final Map<String, BiPredicate<Claim, DecodedJWT>> expectedChecks;
         private long defaultLeeway;
+        private final Map<String, Long> customLeeways;
         private boolean ignoreIssuedAt;
+        private Clock clock;
 
         BaseVerification(Algorithm algorithm) throws IllegalArgumentException {
             if (algorithm == null) {
                 throw new IllegalArgumentException("The Algorithm cannot be null.");
             }
 
             this.algorithm = algorithm;
-            this.expectedClaims = new HashMap<>();
+            this.expectedChecks = new HashMap<>();
+            this.customLeeways = new HashMap<>();
             this.defaultLeeway = 0;
         }
 
         @Override
         public Verification withIssuer(String... issuer) {
-            requireClaim(PublicClaims.ISSUER, isNullOrEmpty(issuer) ? null : Arrays.asList(issuer));
+            List<String> value = isNullOrEmpty(issuer) ? null : Arrays.asList(issuer);
+            checkIfNeedToRemove(PublicClaims.ISSUER, value, ((claim, decodedJWT) -> {
+                if (value == null || !value.contains(claim.asString())) {
+                    throw new InvalidClaimException("The Claim 'iss' value doesn't match the required issuer.");
+                }
+                return true;
+            }));
             return this;
         }
 
         @Override
         public Verification withSubject(String subject) {
-            requireClaim(PublicClaims.SUBJECT, subject);
+            checkIfNeedToRemove(PublicClaims.SUBJECT, subject, (claim, decodedJWT) -> subject.equals(claim.asString()));
```

Comment:
This code will fail with NPE if null is passed as subject. Is it the desired behavior?

Reply:
Since `checkIfNeedToRemove` will check for `null` value for the parameter `subject` and remove the check if it is null, I guess there shouldn't be an NPE issue here